samples/cpp/hello_reshape_ssd/main.cpp

```diff
@@ -2,214 +2,149 @@
-        }
-        if (output_shape.size() != 4) {
-            throw std::logic_error("Incorrect output dimensions for SSD model");
-        }
-        if (output_info == nullptr) {
-            IE_THROW() << "[SAMPLES] internal error - output information is empty";
-        }
+        std::shared_ptr<unsigned char> image_data = reader->getData();
+        size_t image_channels = 3;
+        size_t image_width = reader->width();
+        size_t image_height = reader->height();
 
-        output_info->setPrecision(Precision::FP32);
-
-        auto dumpVec = [](const SizeVector& vec) -> std::string {
-            if (vec.empty())
-                return "[]";
-            std::stringstream oss;
-            oss << "[" << vec[0];
-            for (size_t i = 1; i < vec.size(); i++)
-                oss << "," << vec[i];
-            oss << "]";
-            return oss.str();
-        };
-        std::cout << "Resulting input shape = " << dumpVec(input_shape) << std::endl;
-        std::cout << "Resulting output shape = " << dumpVec(output_shape) << std::endl;
-        // -----------------------------------------------------------------------------------------------------
+        // reshape model to image size and batch size
+        ov::Shape tensor_shape = model->input().get_shape();
 
-        // --------------------------- Step 4. Loading a model to the device
-        // ------------------------------------------
-        ExecutableNetwork executable_network = ie.LoadNetwork(network, device_name);
-        // -----------------------------------------------------------------------------------------------------
+        tensor_shape[0] = batch_size;
+        tensor_shape[1] = image_channels;
+        tensor_shape[2] = image_height;
+        tensor_shape[3] = image_width;
 
-        // --------------------------- Step 5. Create an infer request
+        std::cout << "Reshape network to the image size = [" << image_height << "x" << image_width << "] "
+                  << "with batch = " << batch_size << std::endl;
+        model->reshape({{model->input().get_any_name(), tensor_shape}});
+        printInputAndOutputsInfo(*model);
+
+        // Step 4. Apply preprocessing
+        const ov::Layout tensor_layout{"NHWC"};
+
+        // clang-format off
+        model = PrePostProcessor(model).
+            // 1) InputInfo() with no args assumes a model has a single input
+            input(InputInfo().
+                // 2) Set input tensor information:
+                // - precision of tensor is supposed to be 'u8'
+                // - layout of data is 'NHWC'
+                // - set static spatial dimensions to input tensor to resize from
+                tensor(InputTensorInfo().
+                    set_element_type(ov::element::f32).
+                    set_spatial_static_shape(
+                        tensor_shape[ov::layout::height_idx(tensor_layout)],
+                        tensor_shape[ov::layout::width_idx(tensor_layout)]).
```

Comment:
no resize in this sample, do we don't have to specify `set_spatial_static_shape`. Secondly, comments states about `u8` tensor while actual type is `f32` - which one is right?